### ISSUE TITLE
shebang filter does not behave as expected

### ISSUE THREAD

**Issue by finn** [houseabsolute/perl-code-tidyall#55]
cc @timgimyee / @autarch --

Currently the behavior of the 'shebang' config option does not behave as described in the documentation.

The documentation reads:

> One or more words, separated by whitespace or on multiple lines, indicating
> which shebang lines to accept. This is optional and further filters `select`.

However, if I specify the following in my `.tidyallrc`:

```
[PerlTidy]
select = **/*
ignore = **/*.*
shebang = perl perl5
argv = --profile=.../.perltidyrc
```

And then run:

```
tidyall -l -r bin extlib lib scripts t
```

Expected result: is that the list of files to be perltidied would be every file below bin, extlib, lib, scripts, t that DOES NOT have an extension AND DOES have 'perl' or 'perl5' in the shebang line.

Actual result: the list of files is every file below bin, extlib, lib, scripts, t that does not have an extension.

i.e. the shebang filtering is not actually applied.

Here is the list of file types that were actually matched (generated using the `file` command):

```
ASCII C++ program text
ASCII English text
ASCII c program text
ASCII text
ASCII text, with very long lines
HTML document text
JPEG image data, JFIF standard 1.02
POSIX shell script text executable
a /usr/bin/perl script text executable
a perl -w script text executable
a perl script text executable
a python script text executable
perl Storable (v0.7) data (network-ordered) (major 2) (minor 8)
```

From my testing it appears the shebang regular expression matching is correct (https://github.com/houseabsolute/perl-code-tidyall/blob/master/lib/Code/TidyAll.pm#L457-L458) so I assume the bug lies elsewhere in the `find_matched_files` method (https://github.com/houseabsolute/perl-code-tidyall/blob/master/lib/Code/TidyAll.pm#L444-L472). However, I have not walked through the logic closely enough to determine if the problem is there or elsewhere in `TidyAll.pm`. I decided to file this issue in the hopes that the location of the actual bug would be more apparent to someone more familiar with the codebase.

(I may try to dig into this more on my own at a later date.)

**Comment by autarch** [comment]
A test case would be very helpful. I'm looking at the code and it seems like it should do what you describe.

**Comment by finn** [comment]
Yes agreed -- I looked over the git test code and I think it should be straightforward to add some test cases that cover this. I'm still planning on it I just need to set aside the time to do so.

**Comment by finn** [comment]
Whoops got my issues / PRs mixed up here -- was thinking of the git stash branch. I will see if I can find a test case to reproduce this too.

**Comment by oschwald** [comment]
I also ran into this. The `shebang` filter does not seem to work with the precommit check. All files are checked rather than just ones with the shebang.
